segmentation fault when operational data callback takes more time that rpc timeout

The default sysrepo timeouts are:

```c
/** default timeout for operational subscription callback (ms) */
#define SR_OPER_CB_TIMEOUT 5000
/** default timeout for RPC/action subscription callback (ms) */
#define SR_RPC_CB_TIMEOUT 2000
```

Since commits https://github.com/CESNET/netopeer2/commit/d1e0217a606e8719f6479b340479529b391740ad and https://github.com/CESNET/netopeer2/commit/7c5b112235449ba7ae7d6da6bdbbf0746d1f5da0, `netopeer2-server` uses the default sysrepo timeouts unless the `-t` argument is specified.

When a client connects and sends a `<get>` RPC, the call graph is as follows:

```
MAIN THREAD

np2srv_rpc_cb  "<get>"
  |- sr_rpc_send_tree                    (timeout=2s)
  |   `- sysrepo code which notifies a second thread
  `- returns error to client, client closes session

SECOND THREAD

sr_process_events
 `- np2srv_rpc_get_cb
     |- np2srv_get_rpc_data              (timeout=5s)
     |- np_get_nc_sess_user              (session has been closed by client)
     `- ncac_check_data_read_filter      (with user=NULL)
         `- getpwnam_r(NULL)             segmentation fault
```

How to reproduce:

```console
foo@bar:~$ docker run -it --name sysrepo --rm sysrepo/sysrepo-netopeer2:latest bash -li
...
Digest: sha256:266d2cdfe2c34075c8921b98303322f51e3a7348d8776783f99531d5a3aa0563
Status: Downloaded newer image for sysrepo/sysrepo-netopeer2:latest
root@e450e566bc54:/# gcc application.c -Wall -Wextra -Werror -o application -lyang -lsysrepo
root@e450e566bc54:/# gcc client.c -Wall -Wextra -Werror -o client -lyang -lnetconf2
root@e450e566bc54:/# sysrepoctl -i example.yang
root@e450e566bc54:/# ulimit -c unlimited
root@e450e566bc54:/# ./application &
[1] 88
root@e450e566bc54:/# netopeer2-server -U -d -v 1
[2] 181
root@e450e566bc54:/# ./client 
[WRN]: NP: Session 1 uses a transport protocol not supported by ietf-netconf-monitoring, will not be monitored.
warning: Session 1: unable to identify revision of the schema "ietf-datastores" from the available server side information.
warning: Session 1: unable to identify revision of the schema "ietf-datastores" from the available server side information.
[ERR]: SR: Callback event "rpc" with ID 2 processing timed out.
[WRN]: SR: Event "rpc" with ID 2 priority 0 failed (Timeout expired).
[ERR]: NP: Failed to send an RPC (User callback failed).
[ERR]: SR: Internal error (/usr/src/packages/BUILD/src/shm_mod.c:703).
[ERR]: SR: Internal error (/usr/src/packages/BUILD/src/shm_mod.c:703).
root@e450e566bc54:/# 
[2]+  Segmentation fault      (core dumped) netopeer2-server -U -d -v 1
root@e450e566bc54:/# gdb /usr/bin/netopeer2-server '!usr!bin!netopeer2-server.core'
GNU gdb (Ubuntu 8.1.1-0ubuntu1) 8.1.1
Copyright (C) 2018 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /usr/bin/netopeer2-server...(no debugging symbols found)...done.

warning: core file may not match specified executable file.
[New LWP 182]
[New LWP 183]
[New LWP 184]
[New LWP 185]
[New LWP 181]
[New LWP 186]
[New LWP 187]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `netopeer2-server -U -d -v 1'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  _nss_compat_getpwnam_r (name=0x0, pwd=0x7f5a6998f910, buffer=0x7f5a48017300 "\200S\001HZ\177", buflen=1024, 
    errnop=0x7f5a69990660) at nss_compat/compat-pwd.c:810
810     nss_compat/compat-pwd.c: No such file or directory.
[Current thread is 1 (Thread 0x7f5a69990700 (LWP 182))]
(gdb) bt
#0  _nss_compat_getpwnam_r (name=0x0, pwd=0x7f5a6998f910, buffer=0x7f5a48017300 "\200S\001HZ\177", buflen=1024, 
    errnop=0x7f5a69990660) at nss_compat/compat-pwd.c:810
#1  0x00007f5a6beee72f in __getpwnam_r (name=0x0, resbuf=0x7f5a6998f910, buffer=0x7f5a48017300 "\200S\001HZ\177", 
    buflen=1024, result=0x7f5a6998f908) at ../nss/getXXbyYY_r.c:315
#2  0x000055974fc9c7ab in ncac_getpwnam ()
#3  0x000055974fc9d3ef in ncac_allowed_tree ()
#4  0x000055974fc9f05b in ncac_check_data_read_filter ()
#5  0x000055974fc99ee5 in np2srv_rpc_get_cb ()
#6  0x00007f5a6c230e68 in ?? () from /usr/lib/x86_64-linux-gnu/libsysrepo.so.5
#7  0x00007f5a6c235748 in ?? () from /usr/lib/x86_64-linux-gnu/libsysrepo.so.5
#8  0x00007f5a6c208fb7 in sr_process_events () from /usr/lib/x86_64-linux-gnu/libsysrepo.so.5
#9  0x00007f5a6c236402 in ?? () from /usr/lib/x86_64-linux-gnu/libsysrepo.so.5
#10 0x00007f5a6c78c6db in start_thread (arg=0x7f5a69990700) at pthread_create.c:463
#11 0x00007f5a6bf2c71f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
```

Source files:

```yang
/* example.yang */
module example {
  namespace "urn:example";
  prefix ex;

  container data {
    config false;
    leaf val {
      type string;
    }
  }
}
```

```c
/* application.c */
#include <unistd.h>
#include <signal.h>
#include <sysrepo.h>

int oper_data_cb(
        sr_session_ctx_t *session,
        const char *module_name, const char *path, const char *request_xpath,
        uint32_t request_id, struct lyd_node **parent, void *private_data)
{
        const struct ly_ctx *ctx = sr_get_context(
                sr_session_get_connection(session));

        (void)module_name;
        (void)path;
        (void)request_xpath;
        (void)request_id;
        (void)private_data;

        sleep(3);

        if (*parent) {
                lyd_new_path(*parent, ctx, "/example:data/val",
                        "foobar", 0, 0);
        } else {
                *parent = lyd_new_path(
                        NULL, ctx, "/example:data/val", "foobar",
                        0, 0);
        }

        return SR_ERR_OK;
}

volatile int forever = 1;
void sighandler(int signum)
{
        (void)signum;
        forever = 0;
}

int main(void)
{
        sr_subscription_ctx_t *subscription = NULL;
        sr_session_ctx_t *session = NULL;
        sr_conn_ctx_t *conn = NULL;
        int rc;

        signal(SIGINT, sighandler);
        sr_log_stderr(SR_LL_WRN);

        rc = sr_connect(SR_CONN_DEFAULT, &conn);
        if (rc)
                goto end;

        rc = sr_session_start(conn, SR_DS_RUNNING, &session);
        if (rc)
                goto end;

        rc = sr_oper_get_items_subscribe(
                session, "example", "/example:data",
                oper_data_cb, NULL, 0, &subscription);
        if (rc)
                goto end;

        while (forever)
                sleep(5);

end:
        if (subscription)
                sr_unsubscribe(subscription);
        if (session)
                sr_session_stop(session);
        if (conn)
                sr_disconnect(conn);
        return rc;
}
```

```c
/* client.c */
#include <errno.h>
#include <nc_client.h>

static void nc_log_cb(NC_VERB_LEVEL nc_level, const char *msg)
{
        switch (nc_level) {
        case NC_VERB_ERROR:
                printf("error: %s\n", msg);
                break;
        case NC_VERB_WARNING:
                printf("warning: %s\n", msg);
                break;
        default:
                errno = ECANCELED;
        }
}

int main(void)
{
        struct nc_session *session = NULL;
        struct nc_reply *reply = NULL;
        struct nc_rpc *rpc = NULL;
        NC_MSG_TYPE msgtype;
        uint64_t msgid;
        int ret = 1;

        nc_set_print_clb(nc_log_cb);
        nc_verbosity(NC_VERB_WARNING);

        session = nc_connect_unix("/var/run/netopeer2-server.sock", NULL);
        if (!session)
                goto end;

        rpc = nc_rpc_get("/example:data//.", NC_WD_EXPLICIT, NC_PARAMTYPE_CONST);
        if (!rpc)
                goto end;

        msgtype = nc_send_rpc(session, rpc, -1, &msgid);
        if (msgtype != NC_MSG_RPC)
                goto end;

        msgtype = nc_recv_reply(session, rpc, msgid, -1, 0, &reply);
        if (msgtype == NC_MSG_REPLY) {
                struct nc_reply_data *rdata = (struct nc_reply_data *)reply;
                lyd_print_file(stdout, rdata->data, LYD_XML, LYP_FORMAT);
                ret = 0;
        } else if (msgtype == NC_MSG_ERROR) {
                struct nc_reply_error *err = (struct nc_reply_error *)reply;
                for (uint32_t i = 0; i < err->count; i++) {
                        const struct nc_err *e = &err->err[i];
                        printf("error: %s\n", e->message);
                }
        } else {
                printf("error: invalid reply from NETCONF server\n");
        }

end:
        if (reply)
                nc_reply_free(reply);
        if (rpc)
                nc_rpc_free(rpc);
        if (session)
                nc_session_free(session, NULL);
        return ret;
}
```

Hi Michal,

Segmentation fault again after changing `NC_SESSION_FREE_LOCK_TIMEOUT` with netopeer 1.1.54:

```console
root@73752f61529a:/opt/dev/libnetconf2# git log -n3
commit 0722eb1f43e84eb9152ac0d5fbcfab6293fce573 (HEAD -> devel, tag: v1.1.36, origin/master, origin/devel, origin/HEAD, master)
Author: Michal Vasko <mvasko@cesnet.cz>
Date:   Wed Jan 27 16:12:01 2021 +0100

    packages BUGFIX exact main package version dependency

commit d771fd9c8d9e4a91a174de3af55b2ca100d16b9c
Author: Michal Vasko <mvasko@cesnet.cz>
Date:   Wed Jan 13 14:13:38 2021 +0100

    VERSION bump to version 1.1.36

commit 6087a5265b4f836d1610d826604d5bef0ae08dd3
Author: Michal Vasko <mvasko@cesnet.cz>
Date:   Wed Jan 13 14:13:21 2021 +0100

    SOVERSION bump to version 1.3.15
root@73752f61529a:/opt/dev/libnetconf2# vim src/session_p.h 
root@73752f61529a:/opt/dev/libnetconf2# git diff
diff --git a/src/session_p.h b/src/session_p.h
index 4e1346e..a3b9941 100644
--- a/src/session_p.h
+++ b/src/session_p.h
@@ -314,7 +314,7 @@ struct nc_server_opts {
 /**
  * Timeout in msec for acquiring a lock of a session that is supposed to be freed.
  */
-#define NC_SESSION_FREE_LOCK_TIMEOUT 1000
+#define NC_SESSION_FREE_LOCK_TIMEOUT 5000
 
 /**
  * Timeout in msec for acquiring a lock of a pollsession structure.
root@73752f61529a:/opt/dev/libnetconf2# 
root@73752f61529a:/opt/dev/libnetconf2# make -C build install
make: Entering directory '/opt/dev/libnetconf2/build'
make[1]: Entering directory '/opt/dev/libnetconf2/build'
make[2]: Entering directory '/opt/dev/libnetconf2/build'
make[2]: Leaving directory '/opt/dev/libnetconf2/build'
[  7%] Built target compat
make[2]: Entering directory '/opt/dev/libnetconf2/build'
Scanning dependencies of target netconf2
make[2]: Leaving directory '/opt/dev/libnetconf2/build'
make[2]: Entering directory '/opt/dev/libnetconf2/build'
[ 14%] Building C object CMakeFiles/netconf2.dir/src/io.c.o
[ 21%] Building C object CMakeFiles/netconf2.dir/src/log.c.o
[ 28%] Building C object CMakeFiles/netconf2.dir/src/messages_client.c.o
[ 35%] Building C object CMakeFiles/netconf2.dir/src/messages_server.c.o
[ 42%] Building C object CMakeFiles/netconf2.dir/src/session.c.o
[ 50%] Building C object CMakeFiles/netconf2.dir/src/session_client.c.o
[ 57%] Building C object CMakeFiles/netconf2.dir/src/session_server.c.o
/opt/dev/libnetconf2/src/session_server.c: In function 'nc_accept':
/opt/dev/libnetconf2/src/session_server.c:1757:8: warning: 'uid' may be used uninitialized in this function [-Wmaybe-uninitialized]
     pw = getpwuid(uid);
     ~~~^~~~~~~~~~~~~~~
/opt/dev/libnetconf2/src/session_server.c:1750:11: note: 'uid' was declared here
     uid_t uid;
           ^~~
[ 64%] Building C object CMakeFiles/netconf2.dir/src/time.c.o
[ 71%] Building C object CMakeFiles/netconf2.dir/src/session_client_ssh.c.o
[ 78%] Building C object CMakeFiles/netconf2.dir/src/session_server_ssh.c.o
[ 85%] Building C object CMakeFiles/netconf2.dir/src/session_client_tls.c.o
[ 92%] Building C object CMakeFiles/netconf2.dir/src/session_server_tls.c.o
[100%] Linking C shared library libnetconf2.so
make[2]: Leaving directory '/opt/dev/libnetconf2/build'
[100%] Built target netconf2
make[1]: Leaving directory '/opt/dev/libnetconf2/build'
Install the project...
-- Install configuration: "Release"
-- Installing: /usr/lib/x86_64-linux-gnu/libnetconf2.so.1.3.15
-- Up-to-date: /usr/lib/x86_64-linux-gnu/libnetconf2.so.1
-- Up-to-date: /usr/lib/x86_64-linux-gnu/libnetconf2.so
-- Up-to-date: /usr/include/nc_client.h
-- Up-to-date: /usr/include/nc_server.h
-- Up-to-date: /usr/include/libnetconf2/config.h
-- Up-to-date: /usr/include/libnetconf2/log.h
-- Up-to-date: /usr/include/libnetconf2/netconf.h
-- Up-to-date: /usr/include/libnetconf2/session.h
-- Up-to-date: /usr/include/libnetconf2/messages_client.h
-- Up-to-date: /usr/include/libnetconf2/messages_server.h
-- Up-to-date: /usr/include/libnetconf2/session_client.h
-- Up-to-date: /usr/include/libnetconf2/session_client_ch.h
-- Up-to-date: /usr/include/libnetconf2/session_server.h
-- Up-to-date: /usr/include/libnetconf2/session_server_ch.h
-- Up-to-date: /usr/lib/x86_64-linux-gnu/pkgconfig/libnetconf2.pc
make: Leaving directory '/opt/dev/libnetconf2/build'
root@73752f61529a:/opt/dev/libnetconf2# 
root@73752f61529a:/opt/dev/libnetconf2# ldconfig
root@73752f61529a:/opt/dev/libnetconf2# cd ../Netopeer2
root@73752f61529a:/opt/dev/Netopeer2# git checkout afbaa0e0e4132e4d90d65e5a96ecd9bdab93f9ce
HEAD is now at afbaa0e VERSION bump to version 1.1.54
root@73752f61529a:/opt/dev/Netopeer2# make -C build/ install
...
root@73752f61529a:/opt/dev/Netopeer2# cd
root@73752f61529a:~# ./application &
[1] 747
root@73752f61529a:~# valgrind --track-origins=yes netopeer2-server -U -d &
[2] 1315
root@73752f61529a:~# ==1315== Memcheck, a memory error detector
==1315== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==1315== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==1315== Command: netopeer2-server -U -d
==1315== 
root@73752f61529a:~# ./client 
==1315== Thread 6:
==1315== Conditional jump or move depends on uninitialised value(s)
==1315==    at 0x4C34CF9: strlen (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==1315==    by 0x5B49AED: strdup (strdup.c:41)
==1315==    by 0x4E62C09: nc_sock_accept_binds (in /usr/lib/x86_64-linux-gnu/libnetconf2.so.1.3.15)
==1315==    by 0x4E66895: nc_accept (in /usr/lib/x86_64-linux-gnu/libnetconf2.so.1.3.15)
==1315==    by 0x111539: worker_thread (in /usr/bin/netopeer2-server)
==1315==    by 0x530B6DA: start_thread (pthread_create.c:463)
==1315==    by 0x5BCD71E: clone (clone.S:95)
==1315==  Uninitialised value was created by a stack allocation
==1315==    at 0x4E628BD: nc_sock_accept_binds (in /usr/lib/x86_64-linux-gnu/libnetconf2.so.1.3.15)
==1315== 
==1315== Conditional jump or move depends on uninitialised value(s)
==1315==    at 0x4C34CF9: strlen (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==1315==    by 0x553825C: lydict_insert_zc (in /usr/lib/x86_64-linux-gnu/libyang.so.1.10.7)
==1315==    by 0x4E668F1: nc_accept (in /usr/lib/x86_64-linux-gnu/libnetconf2.so.1.3.15)
==1315==    by 0x111539: worker_thread (in /usr/bin/netopeer2-server)
==1315==    by 0x530B6DA: start_thread (pthread_create.c:463)
==1315==    by 0x5BCD71E: clone (clone.S:95)
==1315==  Uninitialised value was created by a stack allocation
==1315==    at 0x4E628BD: nc_sock_accept_binds (in /usr/lib/x86_64-linux-gnu/libnetconf2.so.1.3.15)
==1315== 
warning: Session 1: unable to identify revision of the schema "ietf-datastores" from the available server side information.
warning: Session 1: unable to identify revision of the schema "ietf-datastores" from the available server side information.
[ERR]: NP: Failed to send an RPC (User callback failed).
root@73752f61529a:~# ==1315== Thread 2:
==1315== Invalid read of size 1
==1315==    at 0x82D90B6: _nss_compat_getpwnam_r (compat-pwd.c:810)
==1315==    by 0x5B8F72E: getpwnam_r@@GLIBC_2.2.5 (getXXbyYY_r.c:315)
==1315==    by 0x11591A: ncac_getpwnam (in /usr/bin/netopeer2-server)
==1315==    by 0x11655E: ncac_allowed_tree (in /usr/bin/netopeer2-server)
==1315==    by 0x1181CA: ncac_check_data_read_filter (in /usr/bin/netopeer2-server)
==1315==    by 0x11351F: np2srv_rpc_get_cb (in /usr/bin/netopeer2-server)
==1315==    by 0x5880E67: ??? (in /usr/lib/x86_64-linux-gnu/libsysrepo.so.5.6.19)
==1315==    by 0x5885747: ??? (in /usr/lib/x86_64-linux-gnu/libsysrepo.so.5.6.19)
==1315==    by 0x5858FB6: sr_process_events (in /usr/lib/x86_64-linux-gnu/libsysrepo.so.5.6.19)
==1315==    by 0x5886401: ??? (in /usr/lib/x86_64-linux-gnu/libsysrepo.so.5.6.19)
==1315==    by 0x530B6DA: start_thread (pthread_create.c:463)
==1315==    by 0x5BCD71E: clone (clone.S:95)
==1315==  Address 0x0 is not stack'd, malloc'd or (recently) free'd
==1315== 
==1315== 
==1315== Process terminating with default action of signal 11 (SIGSEGV): dumping core
==1315==  Access not within mapped region at address 0x0
==1315==    at 0x82D90B6: _nss_compat_getpwnam_r (compat-pwd.c:810)
==1315==    by 0x5B8F72E: getpwnam_r@@GLIBC_2.2.5 (getXXbyYY_r.c:315)
==1315==    by 0x11591A: ncac_getpwnam (in /usr/bin/netopeer2-server)
==1315==    by 0x11655E: ncac_allowed_tree (in /usr/bin/netopeer2-server)
==1315==    by 0x1181CA: ncac_check_data_read_filter (in /usr/bin/netopeer2-server)
==1315==    by 0x11351F: np2srv_rpc_get_cb (in /usr/bin/netopeer2-server)
==1315==    by 0x5880E67: ??? (in /usr/lib/x86_64-linux-gnu/libsysrepo.so.5.6.19)
==1315==    by 0x5885747: ??? (in /usr/lib/x86_64-linux-gnu/libsysrepo.so.5.6.19)
==1315==    by 0x5858FB6: sr_process_events (in /usr/lib/x86_64-linux-gnu/libsysrepo.so.5.6.19)
==1315==    by 0x5886401: ??? (in /usr/lib/x86_64-linux-gnu/libsysrepo.so.5.6.19)
==1315==    by 0x530B6DA: start_thread (pthread_create.c:463)
==1315==    by 0x5BCD71E: clone (clone.S:95)
==1315==  If you believe this happened as a result of a stack
==1315==  overflow in your program's main thread (unlikely but
==1315==  possible), you can try to increase the size of the
==1315==  main thread stack using the --main-stacksize= flag.
==1315==  The main thread stack size used in this run was 8388608.
==1315== 
==1315== HEAP SUMMARY:
==1315==     in use at exit: 799,850 bytes in 6,584 blocks
==1315==   total heap usage: 55,238 allocs, 48,654 frees, 34,047,703 bytes allocated
==1315== 
==1315== LEAK SUMMARY:
==1315==    definitely lost: 4 bytes in 1 blocks
==1315==    indirectly lost: 0 bytes in 0 blocks
==1315==      possibly lost: 1,824 bytes in 6 blocks
==1315==    still reachable: 798,022 bytes in 6,577 blocks
==1315==         suppressed: 0 bytes in 0 blocks
==1315== Rerun with --leak-check=full to see details of leaked memory
==1315== 
==1315== For counts of detected and suppressed errors, rerun with: -v
==1315== ERROR SUMMARY: 3 errors from 3 contexts (suppressed: 0 from 0)

[2]+  Segmentation fault      valgrind --track-origins=yes netopeer2-server -U -d
```

--- a/src/netconf.c
+++ b/src/netconf.c
@@ -185,7 +185,7 @@ cleanup:
 }
 
 int
-np2srv_rpc_get_cb(sr_session_ctx_t *session, const char *op_path, const struct lyd_node *input, sr_event_t UNUSED(event),
+np2srv_rpc_get_cb(sr_session_ctx_t *session, const char *op_path, const struct lyd_node *input, sr_event_t event,
         uint32_t UNUSED(request_id), struct lyd_node *output, void *UNUSED(private_data))
 {
     struct lyd_node *node, *data_get = NULL;
@@ -194,6 +194,11 @@ np2srv_rpc_get_cb(sr_session_ctx_t *session, const char *op_path, const struct l
     struct ly_set *nodeset;
     sr_datastore_t ds = 0;
     char *username = NULL;
+
+    if (event == SR_EV_ABORT) {
+        /* ignore in this case */
+        return SR_ERR_OK;
+    }
 
     /* Get username. It is assumed that right now the NETCONF session cannot end
      * due to the RPC lock held while np2srv_rpc_cb() is executing (which called this callback).
@@ -286,7 +291,7 @@ cleanup:
 
 int
 np2srv_rpc_editconfig_cb(sr_session_ctx_t *session, const char *UNUSED(op_path), const struct lyd_node *input,
-        sr_event_t UNUSED(event), uint32_t UNUSED(request_id), struct lyd_node *UNUSED(output), void *UNUSED(private_data))
+        sr_event_t event, uint32_t UNUSED(request_id), struct lyd_node *UNUSED(output), void *UNUSED(private_data))
 {
     sr_datastore_t ds = 0;
     struct ly_set *nodeset;
@@ -294,6 +299,11 @@ np2srv_rpc_editconfig_cb(sr_session_ctx_t *session, const char *UNUSED(op_path),
     const sr_error_info_t *err_info;
     const char *str, *defop = "merge", *testop = "test-then-set";
     int rc = SR_ERR_OK;
+
+    if (event == SR_EV_ABORT) {
+        /* ignore in this case */
+        return SR_ERR_OK;
+    }
 
     /* get know which datastore is being affected */
     nodeset = lyd_find_path(input, "target/*");
@@ -391,7 +401,7 @@ cleanup:
 
 int
 np2srv_rpc_copyconfig_cb(sr_session_ctx_t *session, const char *UNUSED(op_path), const struct lyd_node *input,
-        sr_event_t UNUSED(event), uint32_t UNUSED(request_id), struct lyd_node *UNUSED(output), void *UNUSED(private_data))
+        sr_event_t event, uint32_t UNUSED(request_id), struct lyd_node *UNUSED(output), void *UNUSED(private_data))
 {
     sr_datastore_t ds = SR_DS_OPERATIONAL, sds = SR_DS_OPERATIONAL;
     struct ly_set *nodeset;
@@ -404,6 +414,11 @@ np2srv_rpc_copyconfig_cb(sr_session_ctx_t *session, const char *UNUSED(op_path),
     const char *trg_url = NULL;
     int lyp_wd_flag;
 #endif
+
+    if (event == SR_EV_ABORT) {
+        /* ignore in this case */
+        return SR_ERR_OK;
+    }
 
     /* Get username. It is assumed that right now the NETCONF session cannot end
      * due to the RPC lock held while np2srv_rpc_cb() is executing (which called this callback).
@@ -560,7 +575,7 @@ cleanup:
 
 int
 np2srv_rpc_deleteconfig_cb(sr_session_ctx_t *session, const char *UNUSED(op_path), const struct lyd_node *input,
-        sr_event_t UNUSED(event), uint32_t UNUSED(request_id), struct lyd_node *UNUSED(output), void *UNUSED(private_data))
+        sr_event_t event, uint32_t UNUSED(request_id), struct lyd_node *UNUSED(output), void *UNUSED(private_data))
 {
     sr_datastore_t ds = 0;
     struct ly_set *nodeset;
@@ -569,6 +584,11 @@ np2srv_rpc_deleteconfig_cb(sr_session_ctx_t *session, const char *UNUSED(op_path
     struct lyd_node *config;
     const char *trg_url = NULL;
 #endif
+
+    if (event == SR_EV_ABORT) {
+        /* ignore in this case */
+        return SR_ERR_OK;
+    }
 
     /* get know which datastore is affected */
     nodeset = lyd_find_path(input, "target/*");
@@ -621,12 +641,17 @@ cleanup:
 
 int
 np2srv_rpc_un_lock_cb(sr_session_ctx_t *session, const char *UNUSED(op_path), const struct lyd_node *input,
-        sr_event_t UNUSED(event), uint32_t UNUSED(request_id), struct lyd_node *UNUSED(output), void *UNUSED(private_data))
+        sr_event_t event, uint32_t UNUSED(request_id), struct lyd_node *UNUSED(output), void *UNUSED(private_data))
 {
     sr_datastore_t ds = 0;
     struct ly_set *nodeset;
     const sr_error_info_t *err_info;
     int rc = SR_ERR_OK;
+
+    if (event == SR_EV_ABORT) {
+        /* ignore in this case */
+        return SR_ERR_OK;
+    }
 
     /* get know which datastore is being affected */
     nodeset = lyd_find_path(input, "target/*");
@@ -663,12 +688,17 @@ cleanup:
 
 int
 np2srv_rpc_kill_cb(sr_session_ctx_t *session, const char *UNUSED(op_path), const struct lyd_node *input,
-        sr_event_t UNUSED(event), uint32_t UNUSED(request_id), struct lyd_node *UNUSED(output), void *UNUSED(private_data))
+        sr_event_t event, uint32_t UNUSED(request_id), struct lyd_node *UNUSED(output), void *UNUSED(private_data))
 {
     struct nc_session *kill_sess;
     struct ly_set *nodeset;
     uint32_t kill_sid, i;
     int rc = SR_ERR_OK;
+
+    if (event == SR_EV_ABORT) {
+        /* ignore in this case */
+        return SR_ERR_OK;
+    }
 
     nodeset = lyd_find_path(input, "session-id");
     kill_sid = ((struct lyd_node_leaf_list *)nodeset->set.d[0])->value.uint32;
@@ -704,10 +734,15 @@ cleanup:
 
 int
 np2srv_rpc_commit_cb(sr_session_ctx_t *session, const char *UNUSED(op_path), const struct lyd_node *UNUSED(input),
-        sr_event_t UNUSED(event), uint32_t UNUSED(request_id), struct lyd_node *UNUSED(output), void *UNUSED(private_data))
+        sr_event_t event, uint32_t UNUSED(request_id), struct lyd_node *UNUSED(output), void *UNUSED(private_data))
 {
     int rc = SR_ERR_OK;;
     const sr_error_info_t *err_info;
+
+    if (event == SR_EV_ABORT) {
+        /* ignore in this case */
+        return SR_ERR_OK;
+    }
 
     /* update sysrepo session datastore */
     sr_session_switch_ds(session, SR_DS_RUNNING);
@@ -728,8 +763,13 @@ cleanup:
 
 int
 np2srv_rpc_discard_cb(sr_session_ctx_t *session, const char *UNUSED(op_path), const struct lyd_node *UNUSED(input),
-        sr_event_t UNUSED(event), uint32_t UNUSED(request_id), struct lyd_node *UNUSED(output), void *UNUSED(private_data))
+        sr_event_t event, uint32_t UNUSED(request_id), struct lyd_node *UNUSED(output), void *UNUSED(private_data))
 {
+    if (event == SR_EV_ABORT) {
+        /* ignore in this case */
+        return SR_ERR_OK;
+    }
+
     int rc = SR_ERR_OK;
 
     /* update sysrepo session datastore */
@@ -749,12 +789,17 @@ cleanup:
 
 int
 np2srv_rpc_validate_cb(sr_session_ctx_t *session, const char *UNUSED(op_path), const struct lyd_node *input,
-        sr_event_t UNUSED(event), uint32_t UNUSED(request_id), struct lyd_node *UNUSED(output), void *UNUSED(private_data))
+        sr_event_t event, uint32_t UNUSED(request_id), struct lyd_node *UNUSED(output), void *UNUSED(private_data))
 {
     sr_datastore_t ds;
     struct lyd_node *config = NULL;
     struct ly_set *nodeset;
     int rc = SR_ERR_OK;
+
+    if (event == SR_EV_ABORT) {
+        /* ignore in this case */
+        return SR_ERR_OK;
+    }
 
     /* get know which datastore is affected */
     nodeset = lyd_find_path(input, "source/*");

